crates/provider/src/provider.rs

```diff
@@ -995,10 +996,35 @@ impl<N: Network, T: Transport + Clone> Provider<N, T> for RootProvider<N, T> {
     }
 }
 
+/// Admin namespace rpc interface that gives access to several non-standard RPC methods.
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+#[allow(unused, unreachable_pub)]
+pub trait AdminApi {
+    /// Adds the given node record to the peerset.
+    async fn add_peer(&self, record: String) -> TransportResult<bool>;
```

Comment:
we have a few more admin endpoints

https://github.com/paradigmxyz/reth/blob/4d798c7b655255b84cbbcd3af2e8827de81e105e/crates/rpc/rpc-api/src/admin.rs#L8-L8

Reply:
Yeah, I can implement them later.

Reply:
I tried to implement `subscribe_peer_events` as below:
```rust    
#[cfg(feature = "pubsub")]
  async fn subscribe_peer_events(&self) -> TransportResult<Subscription<String>> {
      self.root().pubsub_frontend()?;
      let id = self.client().request("admin_peerEvents", ()).await?;
      self.root().get_subscription(id).await
  }
```
and try to subscribe the peer events in tests by:
```rust
let ws = alloy_rpc_client::WsConnect::new(geth2.ws_endpoint());
let client = RpcClient::connect_pubsub(ws).await.unwrap();
let provider2 = RootProvider::<Ethereum, _>::new(client);
let subscription = provider2.subscribe_peer_events().await.unwrap().into_stream();
```
And run the test with `--all-features` but failed with an error message,
```sh
called `Result::unwrap()` on an `Err` value: ErrorResp(ErrorPayload { code: -32601, message: "the method admin_peerEvents does not exist/is not available", data: None })
```
Don't know how to enable it and I find the methods is not list in https://github.com/ethereum/go-ethereum/blob/f1801a9feda8f81532c92077d2c9a8b785fd699b/internal/web3ext/web3ext.go#L183

Maybe leave this method unimplemented now?